core/src/commonMain/kotlin/com/ustadmobile/core/domain/report/model/ReportFilter2.kt

```diff
@@ -1,57 +1,81 @@
 package com.ustadmobile.core.domain.report.model
 
+import com.ustadmobile.core.MR
+import dev.icerock.moko.resources.StringResource
 import kotlinx.serialization.Serializable
 
-@Serializable
-class ReportFilter2(
-    var reportFilterUid: Int = 0,
-
-    var reportFilterSeriesUid: Int = 0,
-
-    var reportFilterField: Int = FIELD_PERSON_GENDER,
-
-    var reportFilterCondition: Int = 0,
-
-    var reportFilterValue: String? = null,
-
-    var reportFilterDropDownValue: Int = 0,
-
-    var reportFilterValueBetweenX: String? = null,
-
-    var reportFilterValueBetweenY: String? = null,
-) {
+/** Enum representing different comparison types */
+enum class Comparisons(val value: Int) {
```

Comment:
The StringResource should be a field on the Comparison Enum e.g.
```
enum class Comparison(val value: Int, val stringResource: StringResource)
```

No need to create the map, extra function, etc.